static/src/javascripts/boot.js

```diff
@@ -40,48 +40,80 @@ const go = () => {
-                                [],
-                                // webpack needs the require function to be called 'require'
-                                // eslint-disable-next-line no-shadow
-                                require => {
-                                    markTime('enhanced boot');
-                                    require('bootstraps/enhanced/main').bootEnhanced();
-
-                                    if (document.readyState === 'complete') {
-                                        capturePerfTimings();
-                                    } else {
-                                        window.addEventListener(
-                                            'load',
-                                            capturePerfTimings
-                                        );
+        if (config.switches.commercial) {
+            commercialBoot = () =>
+                new Promise(resolve => {
+                    if (config.tests.commercialBaseline === 'variant') {
+                        require.ensure(
+                            [],
+                            // webpack needs the require function to be called 'require'
+                            // eslint-disable-next-line no-shadow
+                            require => {
+                                raven.context(
+                                    { tags: { feature: 'commercial-legacy' } },
+                                    () => {
+                                        markTime('commercial boot');
+                                        require(`bootstraps/commercial-legacy`)();
+                                    }
+                                );
+                            },
+                            'commercial-legacy'
+                        );
+                    } else {
+                        require.ensure(
+                            [],
+                            // eslint-disable-next-line no-shadow
+                            require => {
+                                raven.context(
+                                    { tags: { feature: 'commercial' } },
+                                    () => {
+                                        markTime('commercial boot');
+                                        require(`bootstraps/commercial`)();
                                     }
-                                },
-                                'enhanced'
-                            );
-                        }
-                    });
+                                );
+                            },
+                            'commercial'
+                        );
+                    }
+                    resolve();
                 });
-            },
-            'commercial'
-        );
+        }
+
+        // 3. finally, try enhanced
+        // this is defined here so that webpack's code-splitting also
+        // excludes all the modules bundled in the commercial chunk from this one
```

Comment:
the reason we nested enhanced inside commercial before was to make sure enhanced deps didn't duplicate commercial ones – webpack knew what it had already bundled for commercial. this looks as though it will cerate 3 bundles that don't know about each other, and therefore bundle all deps in each – have i got that wrong?

Reply:
(via @SiAdcock btw 😃)

Reply:
fyi you can check these by running `make compile-javascript` and checking `static/target/javascripts/bundle-analyzer-report.html`

Reply:
looks as though it _is_ creating 3 bundles with duplicated deps...

![screenshot-2018-1-16 webpack bundle analyzer](https://user-images.githubusercontent.com/867233/34989040-de2f998e-fab8-11e7-89c9-9ca5fae0e744.png)